pkg/cloud/gcp/actuators/machine/reconciler.go

```diff
@@ -71,9 +71,14 @@ func (r *Reconciler) create() error {
 
 	// networking
 	var networkInterfaces = []*compute.NetworkInterface{}
+
 	for _, nic := range r.providerSpec.NetworkInterfaces {
+		accessConfigs := []*compute.AccessConfig{}
+		if nic.PublicIP {
+			accessConfigs = append(accessConfigs, &compute.AccessConfig{})
+		}
 		computeNIC := &compute.NetworkInterface{
-			AccessConfigs: []*compute.AccessConfig{{}},
```

Comment:
I will not ask how this is working at the first place. I.e. empty `&compute.AccessConfig{}` item without any additional fields set. Though, +1 for making publicIP optional.

Reply:
Yeah, it just needs to be present for you to get a public IP (works same way in Terraform).

Reply:
Strange world we live in :) Thanks for the explanation.